ports/espressif/supervisor/usb_serial_jtag.c

```diff
@@ -101,21 +101,28 @@ bool usb_serial_jtag_connected(void) {
 }
 
 char usb_serial_jtag_read_char(void) {
-    if (ringbuf_num_filled(&ringbuf) == 0) {
+    if (ringbuf_num_filled(&ringbuf) == 0 && !usb_serial_jtag_ll_rxfifo_data_available()) {
         return -1;
     }
-    char c = ringbuf_get(&ringbuf);
+    char c = -1;
```

Comment:
Perhaps move this up and use it for the return?
It should not make much difference since optimisations are made, but yea.

Reply:
Ya, it could be. I'm not sure what the advantage is though.